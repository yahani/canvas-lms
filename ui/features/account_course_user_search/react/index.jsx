/*
 * Copyright (C) 2015 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import {string, bool, shape, func} from 'prop-types'
import {stringify} from 'qs'
import permissionFilter from './helpers/permissionFilter'
import CoursesStore from './store/CoursesStore'
import TermsStore from './store/TermsStore'
import AccountsTreeStore from './store/AccountsTreeStore'
import UsersStore from './store/UsersStore'
import useImmediate from '@canvas/use-immediate-hook'

const stores = [CoursesStore, TermsStore, AccountsTreeStore, UsersStore]

const AccountCourseUserSearch = props => {
  useImmediate(() => {
    const {accountId, rootAccountId} = props
    stores.forEach(s => {
      s.reset({accountId, rootAccountId})
    })
  }, [])

  function updateQueryParams(params) {
    const query = stringify(params)
    window.history.replaceState(null, null, `?${query}`)
  }

  const {tabList} = props.store.getState()
  const tabs = permissionFilter(tabList.tabs, props.permissions)
  const ActivePane = tabs.length === 1 ? tabs[0].pane : tabs[tabList.selected].pane

  return (
    <ActivePane
      {...props}
      onUpdateQueryParams={updateQueryParams}
      queryParams={tabList.queryParams}
    />
  )
}

AccountCourseUserSearch.propTypes = {
  accountId: string.isRequired,
  rootAccountId: string.isRequired,
  permissions: shape({
    analytics: bool.isRequired,
  }).isRequired,
  store: shape({
    getState: func.isRequired,
  }).isRequired,
}

export default AccountCourseUserSearch
