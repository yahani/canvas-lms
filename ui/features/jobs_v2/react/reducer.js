/*
 * Copyright (C) 2022 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

function setQargs(hash) {
  const params = new URLSearchParams(window.location.search)
  for (const [name, value] of Object.entries(hash)) {
    params.set(name, value)
  }
  window.history.replaceState('', '', `?${params.toString()}`)
}

export function initialState() {
  const params = new URLSearchParams(window.location.search)
  return {
    bucket: params.get('bucket') || 'running',
    group_text: params.get('group_text') || '',
    group_type: params.get('group_type') || 'tag',
    group_order: params.get('group_order') || 'info',
    jobs_order: params.get('jobs_order') || 'info',
    groups: [],
    jobs: [],
    job: null,
    jobs_loading: false,
    jobs_page: parseInt(params.get('jobs_page') || '1', 10),
    jobs_page_count: 1,
    jobs_refresh_nonce: 1,
    groups_loading: false,
    groups_page: parseInt(params.get('groups_page') || '1', 10),
    groups_page_count: 1,
    groups_refresh_nonce: 1,
    scope: params.get('scope') || Object.keys(ENV.jobs_scope_filter)[0],
    auto_refresh: (params.get('auto_refresh') || '0') !== '0',
    start_date: params.get('start_date'),
    end_date: params.get('end_date'),
    time_zone: params.get('time_zone') || ENV?.TIMEZONE || 'UTC',
  }
}

export function jobsReducer(prevState, action) {
  if (action.type === 'CHANGE_BUCKET') {
    setQargs({bucket: action.payload})
    return {
      ...prevState,
      bucket: action.payload,
      groups: [],
      jobs: [],
      job: null,
      groups_page: 1,
      jobs_page: 1,
      groups_page_count: 1,
      jobs_page_count: 1,
    }
  } else if (action.type === 'CHANGE_GROUP_TYPE') {
    setQargs({group_type: action.payload})
    return {
      ...prevState,
      group_type: action.payload,
      groups: [],
      jobs: [],
      job: null,
      groups_page: 1,
      jobs_page: 1,
      groups_page_count: 1,
      jobs_page_count: 1,
    }
  } else if (action.type === 'CHANGE_GROUP_ORDER') {
    setQargs({group_order: action.payload})
    return {...prevState, group_order: action.payload, groups: []}
  } else if (action.type === 'GROUPS_LOADING') {
    return {...prevState, groups_loading: action.payload}
  } else if (action.type === 'REFRESH_GROUPS') {
    return {...prevState, groups_refresh_nonce: prevState.groups_refresh_nonce + 1}
  } else if (action.type === 'FETCHED_GROUPS') {
    return {...prevState, groups: action.payload}
  } else if (action.type === 'GROUP_METADATA') {
    if (action.payload.link) {
      const last = parseInt(action.payload.link.last?.page, 10) || 1
      return {
        ...prevState,
        groups_page_count: last,
        groups_page: Math.min(prevState.groups_page, last),
      }
    }
  } else if (action.type === 'CHANGE_GROUPS_PAGE') {
    setQargs({groups_page: action.payload})
    return {...prevState, groups_page: action.payload}
  } else if (action.type === 'CHANGE_GROUP_TEXT') {
    if (prevState.group_text !== action.payload) {
      setQargs({group_text: action.payload})
      return {
        ...prevState,
        group_text: action.payload,
        jobs: [],
        job: null,
        jobs_page: 1,
        jobs_page_count: 1,
      }
    } else {
      return prevState
    }
  } else if (action.type === 'CHANGE_JOBS_ORDER') {
    setQargs({jobs_order: action.payload})
    return {...prevState, jobs_order: action.payload, jobs: [], job: null}
  } else if (action.type === 'JOBS_LOADING') {
    return {...prevState, jobs_loading: action.payload}
  } else if (action.type === 'REFRESH_JOBS') {
    return {...prevState, jobs_refresh_nonce: prevState.jobs_refresh_nonce + 1}
  } else if (action.type === 'FETCHED_JOBS') {
    const job = action.payload.find(j => j.id === prevState.job?.id) || prevState.job
    return {...prevState, jobs: action.payload, job}
  } else if (action.type === 'JOBS_METADATA') {
    if (action.payload.link) {
      const last = parseInt(action.payload.link.last?.page, 10) || 1
      return {
        ...prevState,
        jobs_page_count: last,
        jobs_page: Math.min(prevState.jobs_page, last),
      }
    }
  } else if (action.type === 'CHANGE_JOBS_PAGE') {
    setQargs({jobs_page: action.payload})
    return {...prevState, jobs_page: action.payload}
  } else if (action.type === 'SELECT_JOB') {
    return {...prevState, job: action.payload}
  } else if (action.type === 'CHANGE_SCOPE') {
    setQargs({jobs_scope: action.payload})
    return {
      ...prevState,
      groups: [],
      jobs: [],
      job: null,
      groups_page: 1,
      jobs_page: 1,
      groups_page_count: 1,
      jobs_page_count: 1,
      scope: action.payload,
    }
  } else if (action.type === 'TOGGLE_AUTO_REFRESH') {
    if (prevState.auto_refresh) {
      setQargs({auto_refresh: '0'})
      return {...prevState, auto_refresh: false}
    } else {
      setQargs({auto_refresh: '1'})
      return {
        ...prevState,
        auto_refresh: true,
        groups_refresh_nonce: prevState.groups_refresh_nonce + 1,
        jobs_refresh_nonce: prevState.jobs_refresh_nonce + 1,
      }
    }
  } else if (action.type === 'REFRESH_ALL') {
    return {
      ...prevState,
      groups_refresh_nonce: prevState.groups_refresh_nonce + 1,
      jobs_refresh_nonce: prevState.jobs_refresh_nonce + 1,
    }
  } else if (action.type === 'CHANGE_DATE_OPTIONS') {
    setQargs({
      start_date: action.payload.start_date || '',
      end_date: action.payload.end_date || '',
      time_zone: action.payload.time_zone,
    })
    if (
      action.payload.start_date !== prevState.start_date ||
      action.payload.end_date !== prevState.end_date
    ) {
      return {
        ...prevState,
        start_date: action.payload.start_date,
        end_date: action.payload.end_date,
        time_zone: action.payload.time_zone,
        groups: [],
        jobs: [],
        groups_page: 1,
        jobs_page: 1,
        groups_page_count: 1,
        jobs_page_count: 1,
      }
    } else {
      return {
        ...prevState,
        time_zone: action.payload.time_zone,
      }
    }
  }
}
