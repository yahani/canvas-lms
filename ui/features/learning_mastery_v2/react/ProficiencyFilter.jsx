/*
 * Copyright (C) 2021 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react'
import PropTypes from 'prop-types'
import {View} from '@instructure/ui-view'
import {useScope as useI18nScope} from '@canvas/i18n'
import ProficiencyRating from './ProficiencyRating'
import {proficiencyRatingShape} from './shapes'

const I18n = useI18nScope('LearningMasteryGradebook')

const ProficiencyFilter = ({ratings, visibleRatings, setVisibleRatings}) => {
  const onClickPill = (isVisible, visibleIndex) => {
    let index
    if (visibleIndex === null) {
      index = 0
    } else {
      index = visibleIndex + 1
    }
    const visible = [...visibleRatings]
    visible[index] = isVisible
    setVisibleRatings(visible)
  }

  return (
    <View display="flex" padding="small 0 small 0">
      <View display="flex">
        {ratings.map(({points, color, description, masteryAt}) => (
          <ProficiencyRating
            points={points}
            masteryAt={masteryAt}
            color={color}
            description={description}
            onClick={onClickPill}
            key={points}
          />
        ))}
      </View>
    </View>
  )
}

ProficiencyFilter.propTypes = {
  ratings: PropTypes.arrayOf(PropTypes.shape(proficiencyRatingShape)).isRequired,
  visibleRatings: PropTypes.arrayOf(PropTypes.bool).isRequired,
  setVisibleRatings: PropTypes.func.isRequired,
}
export default ProficiencyFilter
