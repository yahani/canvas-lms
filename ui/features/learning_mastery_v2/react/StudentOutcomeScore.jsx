/*
 * Copyright (C) 2021 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react'
import PropTypes from 'prop-types'
import {ScreenReaderContent} from '@instructure/ui-a11y-content'
import {Flex} from '@instructure/ui-flex'
import SVGWrapper from '@canvas/svg-wrapper'
import {svgUrl} from './icons'
import {outcomeShape, outcomeRollupShape} from './shapes'
import {useScope as useI18nScope} from '@canvas/i18n'

const I18n = useI18nScope('learning_mastery_gradebook')

const StudentOutcomeScore = ({outcome, rollup, visibleRatings}) => {
  let isVisible

  // If account level mastery scales are disabled, then always
  // show the icons in the outcome score box
  if (!ENV.GRADEBOOK_OPTIONS.ACCOUNT_LEVEL_MASTERY_SCALES) {
    isVisible = true
  } else {
    isVisible = rollup ? visibleRatings[rollup.rating.points + 1] : visibleRatings[0]
  }

  return (
    <Flex width="100%" height="100%" alignItems="center" justifyItems="center">
      {isVisible && (
        <div style={{display: 'flex'}}>
          <SVGWrapper
            fillColor={rollup?.rating?.color}
            url={svgUrl(rollup?.rating?.points, outcome.mastery_points)}
            style={{display: 'flex', alignItems: 'center', justifyItems: 'center', padding: '0px'}}
          />
          <ScreenReaderContent>
            {rollup?.rating?.description || I18n.t('Unassessed')}
          </ScreenReaderContent>
        </div>
      )}
    </Flex>
  )
}

StudentOutcomeScore.propTypes = {
  outcome: PropTypes.shape(outcomeShape).isRequired,
  rollup: PropTypes.shape(outcomeRollupShape),
  visibleRatings: PropTypes.arrayOf(PropTypes.bool).isRequired,
}

export default StudentOutcomeScore
