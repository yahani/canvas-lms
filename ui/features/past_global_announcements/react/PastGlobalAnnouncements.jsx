/*
 * Copyright (C) 2020 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React, {useState} from 'react'
import {Heading} from '@instructure/ui-heading'
import {Tabs} from '@instructure/ui-tabs'
import {Text} from '@instructure/ui-text'
import {View} from '@instructure/ui-view'
import {ScreenReaderContent} from '@instructure/ui-a11y-content'
import AnnouncementFactory from './AnnouncementFactory'
import {useScope as useI18nScope} from '@canvas/i18n'

const I18n = useI18nScope('past_global_announcements')

const PastGlobalAnnouncements = () => {
  const [selectedIndex, setSelectedIndex] = useState(0)
  const activeAnnouncements = AnnouncementFactory(ENV.global_notifications.current, 'Current')
  const pastAnnouncements = AnnouncementFactory(ENV.global_notifications.past, 'Past')

  return (
    <>
      <Heading level="h1">
        <ScreenReaderContent>{I18n.t('Global Announcements')}</ScreenReaderContent>
      </Heading>
      <Tabs
        margin="0 auto"
        padding="medium"
        onRequestTabChange={(_event, {index}) => setSelectedIndex(index)}
      >
        <Tabs.Panel
          id="currentTab"
          renderTitle={I18n.t('Current')}
          isSelected={selectedIndex === 0}
          data-testid="GlobalAnnouncementCurrentTab"
        >
          <View margin="0 0 small 0" display="block">
            <Text size="medium" lineHeight="double">
              {I18n.t('Active Announcements')}
            </Text>
          </View>
          {activeAnnouncements}
        </Tabs.Panel>
        <Tabs.Panel
          id="pastTab"
          renderTitle={I18n.t('Recent')}
          isSelected={selectedIndex === 1}
          data-testid="GlobalAnnouncementPastTab"
        >
          <View margin="0 0 small 0" display="block">
            <Text size="medium" lineHeight="double">
              {I18n.t('Announcements from the past four months')}
            </Text>
          </View>
          {pastAnnouncements}
        </Tabs.Panel>
      </Tabs>
    </>
  )
}

export default PastGlobalAnnouncements
