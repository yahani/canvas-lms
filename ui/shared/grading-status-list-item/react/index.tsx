/*
 * Copyright (C) 2023 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import {View} from '@instructure/ui-view'
import type {BorderWidth} from '@instructure/emotion'
import type {Cursor} from '@instructure/ui-prop-types'

type GradingStatusListItemProps = {
  children: React.ReactNode
  cursor?: Cursor
  backgroundColor: string
  borderColor?:
    | 'transparent'
    | 'primary'
    | 'secondary'
    | 'brand'
    | 'info'
    | 'success'
    | 'warning'
    | 'alert'
    | 'danger'
  borderStyle?: string
  borderWidth?: BorderWidth
  display?: 'auto' | 'inline' | 'block' | 'inline-block' | 'flex' | 'inline-flex'
  setElementRef?: (ref: Element | null) => void
}
export const GradingStatusListItem = ({
  backgroundColor,
  borderColor,
  borderStyle,
  borderWidth,
  children,
  cursor,
  display,
  setElementRef,
}: GradingStatusListItemProps) => {
  return (
    <View
      as="div"
      borderWidth={borderWidth ?? 'medium'}
      borderColor={borderColor ?? 'primary'}
      borderRadius="large"
      cursor={cursor}
      padding="x-small"
      background="primary"
      height="3rem"
      width="16.25rem"
      display={display ?? 'block'}
      elementRef={setElementRef}
      themeOverride={{backgroundPrimary: backgroundColor, borderStyle: borderStyle ?? 'solid'}}
    >
      {children}
    </View>
  )
}
