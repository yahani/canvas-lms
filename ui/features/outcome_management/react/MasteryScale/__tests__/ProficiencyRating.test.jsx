// @vitest-environment jsdom
/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import {mount, shallow} from 'enzyme'
import {render, fireEvent} from '@testing-library/react'
import ProficiencyRating from '../ProficiencyRating'

describe('ProficiencyRating', () => {
  let onDeleteMock
  let onFocusChangeMock
  let onMasteryChangeMock
  let onDescriptionChangeMock
  let onPointsChangeMock

  const defaultProps = (props = {}) => ({
    color: '00ff00',
    description: 'Stellar',
    disableDelete: false,
    mastery: false,
    canManage: false,
    onColorChange: () => {},
    onDelete: onDeleteMock,
    onDescriptionChange: onDescriptionChangeMock,
    onFocusChange: onFocusChangeMock,
    onMasteryChange: onMasteryChangeMock,
    onPointsChange: onPointsChangeMock,
    points: '10.0',
    position: 1,
    ...props,
  })

  beforeEach(() => {
    onDeleteMock = jest.fn()
    onFocusChangeMock = jest.fn()
    onMasteryChangeMock = jest.fn()
    onDescriptionChangeMock = jest.fn()
    onPointsChangeMock = jest.fn()
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('can not manage', () => {
    it('renders the ProficiencyRating component', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps()} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('mastery checkbox is checked if mastery', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps({mastery: true})} />)
      const radio = wrapper.find('RadioInput')
      expect(radio.props().checked).toBe(true)
    })

    it('mastery checkbox does not appear if not mastery', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps()} />)
      const radio = wrapper.find('RadioInput')
      expect(radio.exists()).toBeFalsy()
    })

    it('mastery checkbox does not receive focus', () => {
      const wrapper = mount(
        <div>
          <ProficiencyRating {...defaultProps({focusField: 'mastery', mastery: true})} />
        </div>
      )
      expect(wrapper.find('RadioInput').find('input').instance()).not.toBe(document.activeElement)
    })

    it('clicking mastery checkbox does not trigger change', () => {
      const wrapper = mount(<ProficiencyRating {...defaultProps({mastery: true})} />)
      wrapper.find('RadioInput').find('input').simulate('change')
      expect(onMasteryChangeMock).not.toHaveBeenCalled()
    })

    it('does not render TextInput', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps()} />)
      expect(wrapper.find('TextInput').exists()).toBeFalsy()
    })

    it('does not render delete button', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps()} />)
      expect(wrapper.find('.deleteButton').exists()).toBeFalsy()
    })

    it('includes the points', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps()} />)
      const content = wrapper.find('.points').find('PresentationContent').at(0)
      expect(content.childAt(0).text()).toBe('10')
    })
  })

  describe('can manage', () => {
    it('renders the ProficiencyRating component', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps({canManage: true})} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('mastery checkbox is checked if mastery', () => {
      const wrapper = shallow(
        <ProficiencyRating
          {...defaultProps({
            mastery: true,
            canManage: true,
          })}
        />
      )
      const radio = wrapper.find('RadioInput')
      expect(radio.props().checked).toBe(true)
    })

    it('clicking mastery checkbox triggers change', () => {
      const wrapper = mount(<ProficiencyRating {...defaultProps({canManage: true})} />)
      wrapper.find('RadioInput').find('input').simulate('change')
      expect(onMasteryChangeMock).toHaveBeenCalledTimes(1)
    })

    it('includes the rating description', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps({canManage: true})} />)
      const input = wrapper.find('TextInput').at(0)
      expect(input.prop('defaultValue')).toBe('Stellar')
    })

    it('changing description triggers change', () => {
      const wrapper = mount(<ProficiencyRating {...defaultProps({canManage: true})} />)
      wrapper.find('TextInput').at(0).find('input').simulate('change')
      expect(onDescriptionChangeMock).toHaveBeenCalledTimes(1)
    })

    it('includes the points', () => {
      const wrapper = shallow(<ProficiencyRating {...defaultProps({canManage: true})} />)
      const input = wrapper.find('TextInput').at(1)
      expect(input.prop('defaultValue')).toBe('10')
    })

    it('changing points triggers change', () => {
      const {getAllByRole} = render(<ProficiencyRating {...defaultProps({canManage: true})} />)
      const secondInput = getAllByRole('textbox')[1]
      fireEvent.change(secondInput, {target: {value: 'some new value'}})

      expect(onPointsChangeMock).toHaveBeenCalledTimes(1)
    })

    it('calls onDelete prop when click on delete and confirm in the confirmation modal', () => {
      const {getByText} = render(<ProficiencyRating {...defaultProps({canManage: true})} />)
      fireEvent.click(getByText('Delete mastery level 1'))
      fireEvent.click(getByText('Confirm'))
      expect(onDeleteMock).toHaveBeenCalledTimes(1)
    })

    it('clicking disabled delete button does not show delete modal', () => {
      const {queryByText} = render(
        <ProficiencyRating
          {...defaultProps({
            disableDelete: true,
            canManage: true,
          })}
        />
      )
      fireEvent.click(queryByText('Delete mastery level 1'))
      expect(queryByText('Remove Mastery Level')).not.toBeInTheDocument()
    })

    it('shows color input', () => {
      const {getByText} = render(<ProficiencyRating {...defaultProps({canManage: true})} />)
      expect(getByText('Change color for mastery level 1')).toBeInTheDocument()
    })

    it('calls onFocusChange prop when current input looses focus', () => {
      const wrapper = mount(<ProficiencyRating {...defaultProps({canManage: true})} />)
      wrapper.find('TextInput').at(0).find('input').simulate('blur')
      wrapper.find('TextInput').at(1).find('input').simulate('blur')
      expect(onFocusChangeMock).toHaveBeenCalledTimes(2)
    })

    describe('when individualOutcome is true', () => {
      it('hides color input', () => {
        const {queryByText, container} = render(
          <ProficiencyRating {...defaultProps({canManage: true, individualOutcome: true})} />
        )
        expect(queryByText('Change color for mastery level 1')).not.toBeInTheDocument()
        expect(container.getElementsByClassName('color').length).toBe(0)
      })
    })
  })
})
