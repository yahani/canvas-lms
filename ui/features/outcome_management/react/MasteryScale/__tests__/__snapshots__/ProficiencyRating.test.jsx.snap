// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProficiencyRating can manage renders the ProficiencyRating component 1`] = `
<Flex
  alignItems="start"
  as="span"
  direction="row"
  display="flex"
  gap="none"
  justifyItems="start"
  padding="0 small small small"
  width="100%"
  withVisualDebug={false}
  wrap="no-wrap"
>
  <Item
    as="span"
    padding="0 medium 0 0"
    shouldGrow={false}
    shouldShrink={false}
    size="15%"
    textAlign="center"
  >
    <div
      className="mastery null"
    >
      <RadioInput
        checked={false}
        context="success"
        disabled={false}
        inline={false}
        label={
          <ScreenReaderContent
            as="span"
          >
            Mastery false for mastery level 1
          </ScreenReaderContent>
        }
        onChange={[Function]}
        readOnly={false}
        size="medium"
        variant="simple"
      />
    </div>
  </Item>
  <Item
    align="start"
    as="span"
    padding="0 small 0 0"
    shouldGrow={false}
    shouldShrink={false}
    size="40%"
  >
    <div
      className="description"
    >
      <TextInput
        data-testid="rating-description-input"
        defaultValue="Stellar"
        display="block"
        inputRef={[Function]}
        isRequired={false}
        messages={null}
        onBlur={[Function]}
        onChange={[Function]}
        renderLabel={
          <ScreenReaderContent
            as="span"
          >
            Change description for mastery level 1
          </ScreenReaderContent>
        }
        shouldNotWrap={false}
        size="medium"
        textAlign="start"
        type="text"
        width="100%"
      />
    </div>
  </Item>
  <Item
    align="start"
    as="span"
    padding="0 small 0 0"
    shouldGrow={false}
    shouldShrink={false}
    size="15%"
  >
    <div
      className="points"
    >
      <TextInput
        data-testid="rating-points-input"
        defaultValue="10"
        display="block"
        inputRef={[Function]}
        isRequired={false}
        messages={null}
        onBlur={[Function]}
        onChange={[Function]}
        renderLabel={
          <ScreenReaderContent
            as="span"
          >
            Change points for mastery level 1
          </ScreenReaderContent>
        }
        shouldNotWrap={false}
        size="medium"
        textAlign="start"
        type="text"
        width="4rem"
      />
      <div
        aria-hidden="true"
        className="pointsDescription"
      >
        points
      </div>
    </div>
  </Item>
  <Item
    as="span"
    shouldGrow={false}
    shouldShrink={false}
  >
    <div
      className="color"
    >
      <Popover
        color="primary"
        constrain="window"
        defaultIsShowingContent={false}
        insertAt="bottom"
        isShowingContent={false}
        offsetX={0}
        offsetY={0}
        on="click"
        onHideContent={[Function]}
        onPositioned={[Function]}
        onShowContent={[Function]}
        placement="bottom center"
        renderTrigger={
          <Link
            as="button"
            color="link"
            forceButtonRole={true}
            iconPlacement="start"
            isWithinText={false}
          >
            <div
              style={
                {
                  "margin": "0 0.8rem",
                  "padding": "0.55rem 0 0",
                }
              }
            >
              <span
                className="colorPickerIcon"
                style={
                  {
                    "background": "#00ff00",
                  }
                }
              />
              <ScreenReaderContent
                as="span"
              >
                Change color for mastery level 1
              </ScreenReaderContent>
              <span
                aria-hidden="true"
              >
                Change
              </span>
            </div>
          </Link>
        }
        shadow="resting"
        shouldAlignArrow={false}
        shouldCloseOnDocumentClick={true}
        shouldCloseOnEscape={true}
        shouldContainFocus={true}
        shouldFocusContentOnTriggerBlur={true}
        shouldRenderOffscreen={false}
        shouldReturnFocus={true}
        shouldTrackPosition={true}
        stacking="topmost"
        withArrow={true}
      >
        <ColorPicker
          afterClose={[Function]}
          allowWhite={false}
          colors={
            [
              {
                "hexcode": "#BD3C14",
                "name": "Brick",
              },
              {
                "hexcode": "#FF2717",
                "name": "Red",
              },
              {
                "hexcode": "#E71F63",
                "name": "Magenta",
              },
              {
                "hexcode": "#8F3E97",
                "name": "Purple",
              },
              {
                "hexcode": "#65499D",
                "name": "Deep Purple",
              },
              {
                "hexcode": "#4554A4",
                "name": "Indigo",
              },
              {
                "hexcode": "#1770AB",
                "name": "Blue",
              },
              {
                "hexcode": "#0B9BE3",
                "name": "Light Blue",
              },
              {
                "hexcode": "#06A3B7",
                "name": "Cyan",
              },
              {
                "hexcode": "#009688",
                "name": "Teal",
              },
              {
                "hexcode": "#009606",
                "name": "Green",
              },
              {
                "hexcode": "#8D9900",
                "name": "Olive",
              },
              {
                "hexcode": "#D97900",
                "name": "Pumpkin",
              },
              {
                "hexcode": "#FD5D10",
                "name": "Orange",
              },
              {
                "hexcode": "#F06291",
                "name": "Pink",
              },
            ]
          }
          currentColor="#00ff00"
          focusOnMount={false}
          hideOnScroll={false}
          hidePrompt={true}
          nonModal={true}
          parentComponent="ProficiencyRating"
          setStatusColor={[Function]}
          withAnimation={false}
          withArrow={false}
          withBorder={false}
          withBoxShadow={false}
          withDarkCheck={false}
        />
      </Popover>
    </div>
  </Item>
  <Item
    as="span"
    padding="0 small 0 small"
    shouldGrow={false}
    shouldShrink={false}
    size="10%"
  >
    <div
      className="deleteButton"
    >
      <IconButton
        as="button"
        color="secondary"
        cursor="pointer"
        data-testid="rating-delete-btn"
        disabled={false}
        elementRef={[Function]}
        margin="0"
        onClick={[Function]}
        renderIcon={<IconTrashLine />}
        screenReaderLabel="Delete mastery level 1"
        shape="rectangle"
        size="medium"
        type="button"
        withBackground={false}
        withBorder={false}
      />
      <ConfirmMasteryModal
        confirmButtonText="Confirm"
        isOpen={false}
        modalText="This will remove the mastery level from your mastery scale."
        onClose={[Function]}
        onConfirm={[Function]}
        title="Remove Mastery Level"
      />
    </div>
  </Item>
</Flex>
`;

exports[`ProficiencyRating can not manage renders the ProficiencyRating component 1`] = `
<Flex
  alignItems="start"
  as="span"
  direction="row"
  display="flex"
  gap="none"
  justifyItems="start"
  padding="0 small small small"
  width="100%"
  withVisualDebug={false}
  wrap="no-wrap"
>
  <Item
    as="span"
    padding="0 medium 0 0"
    shouldGrow={false}
    shouldShrink={false}
    size="15%"
    textAlign="center"
  >
    <div
      className="mastery view-only"
    />
  </Item>
  <Item
    align="start"
    as="span"
    padding="0 small 0 0"
    shouldGrow={false}
    shouldShrink={false}
    size="40%"
  >
    <div
      className="description"
    >
      <Text
        as="span"
        letterSpacing="normal"
        size="medium"
        wrap="normal"
      >
        <ScreenReaderContent
          as="span"
        >
          Description for mastery level 1: Stellar
        </ScreenReaderContent>
        <PresentationContent
          as="span"
        >
          Stellar
        </PresentationContent>
      </Text>
    </div>
  </Item>
  <Item
    align="start"
    as="span"
    padding="0 small 0 0"
    shouldGrow={false}
    shouldShrink={false}
    size="15%"
  >
    <div
      className="points"
    >
      <View
        borderColor="primary"
        display="auto"
        focusColor="info"
        focusPosition="offset"
        margin="0 0 0 small"
        overflowX="visible"
        overflowY="visible"
        overscrollBehavior="auto"
        position="static"
        shouldAnimateFocus={true}
        withVisualDebug={false}
      >
        <ScreenReaderContent
          as="span"
        >
          Points for mastery level 1: 10.0
        </ScreenReaderContent>
        <PresentationContent
          as="span"
        >
          10
          <div
            className="pointsDescription view-only"
          >
            points
          </div>
        </PresentationContent>
      </View>
    </div>
  </Item>
  <Item
    as="span"
    shouldGrow={false}
    shouldShrink={false}
  >
    <div
      className="color"
    >
      <span
        className="colorPickerIcon"
        style={
          {
            "background": "#00ff00",
            "marginLeft": "2rem",
          }
        }
      >
        <ScreenReaderContent
          as="span"
        >
          Color #00ff00 for mastery level 1
        </ScreenReaderContent>
      </span>
    </div>
  </Item>
</Flex>
`;
